### PR TITLE
Bing/ft watcher mainnet

### PR DESCRIPTION
**Pull request by bingyuyap**
This PR adds mainnet related configuration and logic for the Fast Transfer Watcher:

**Config**

- Config for `Solana`, `Base` and `Arbitrum`
  - blocks to start observing from on Mainnet
  - Mainnet contract addresses
- Update `example-liquidity-layer-solana` tarball file to have Mainnet program ID for Matching Eng

**Logic**

- Add conditional function selector for Mainnet vs Testnet. This is because on Mainnet the entry point is the Swap Layer but on Testnet it is the Token Router 
- Add logic to handle missing `fast_vaa_hash` in Mainnet and Testnet. 

**Clean up**
- Fix `makeBlockKey` on EVM 
- Improve logging on `FTSolanaWatcher`
- Clean up `makeFinalizedFTWatcher`

### CODE REVIEW

File: watcher/package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "read-firestore": "ts-node scripts/readFirestore.ts",
     "reconstruct-vaa": "ts-node scripts/reconstructVAA.ts",
     "update-found-vaas": "ts-node scripts/updateFoundVAAs.ts",
-    "update-rows": "ts-node scripts/updateRows.ts"
+    "update-rows": "ts-node scripts/updateRows.ts",
+    "watch-ft-sol": "ts-node scripts/watchFtSolana.ts"
```
Comment (bingyuyap) [inline watcher/package.json:28]: remove this

File: watcher/scripts/watchFtSolana.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import * as dotenv from 'dotenv';
```
Comment (bingyuyap) [inline watcher/scripts/watchFtSolana.ts:1]: remove this script

File: watcher/src/fastTransfer/consts.ts [hunk 3]
```diff
@@ -1,11 +1,17 @@
-import { Network } from '@wormhole-foundation/sdk-base';
+import { Chain, Network } from '@wormhole-foundation/sdk-base';
 
 export type FastTransferContracts = 'MatchingEngine' | 'TokenRouter' | 'USDCMint';
 
 // Will define more as we know what the mainnet addresses are
```
Comment (bingyuyap) [inline watcher/src/fastTransfer/consts.ts:5]: remove comment

File: watcher/src/index.ts [hunk 4]
```diff
@@ -93,14 +94,15 @@ const supportedNTTChains: Chain[] =
     ? ['Solana', 'Sepolia', 'ArbitrumSepolia', 'BaseSepolia', 'OptimismSepolia']
     : ['Solana', 'Ethereum', 'Fantom', 'Arbitrum', 'Optimism', 'Base'];
 
-const supportedFTChains: Chain[] = network === 'Testnet' ? ['Solana', 'ArbitrumSepolia'] : [];
+const supportedFTEVMChain: Chain[] =
```
Comment (bingyuyap) [inline watcher/src/index.ts:97]: this should be `supportedFTChain`

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 5]
```diff
@@ -91,6 +92,7 @@ export class FTSolanaWatcher extends SolanaWatcher {
       new PublicKey(this.MATCHING_ENGINE_PROGRAM_ID),
       this.matchingEngineBorshCoder
     );
+    this.isTest = isTest;
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:95]: explain plzz

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 6]
```diff
@@ -949,7 +968,7 @@ export class FTSolanaWatcher extends SolanaWatcher {
       const result = await this.pg('auction_history_mapping')
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:968]: results not used

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 7]
```diff
@@ -969,7 +988,7 @@ export class FTSolanaWatcher extends SolanaWatcher {
         .insert(mappings)
         .onConflict('auction_pubkey')
         .merge();
-      this.logger.info(`Saved ${mappings.length} auction history mappings. Result:`, result);
+      this.logger.info(`Saved ${mappings.length} auction history mappings. Result:`, mappings);
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:991]: not a good idea to log mappings

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 8]
```diff
@@ -987,7 +1006,7 @@ export class FTSolanaWatcher extends SolanaWatcher {
         .insert(fastTransfer)
         .onConflict('fast_vaa_id')
         .merge();
-      this.logger.info(`Saved fast transfer ${fastTransfer.fast_vaa_hash}. Result:`, result);
+      this.logger.info(`Saved fast transfer ${fastTransfer.fast_vaa_hash}. Result:`, fastTransfer);
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:1009]: `fastTransfer` wont be logged here

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 9]
```diff
@@ -1000,13 +1019,14 @@ export class FTSolanaWatcher extends SolanaWatcher {
       return;
     }
     try {
-      const result = await this.pg('market_orders')
-        .insert(update)
-        .onConflict('fast_vaa_hash')
-        .merge();
-      this.logger.info(`Updated market order ${update.fast_vaa_id}. Result:`, result);
+      const conflictKey = update.fast_vaa_id ? 'fast_vaa_id' : 'fast_vaa_hash';
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:1022]: comment to explain

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 10]
```diff
@@ -1064,13 +1084,23 @@ export class FTSolanaWatcher extends SolanaWatcher {
     }
     try {
       const result = await this.pg(table).insert(info);
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:1086]: should just log the id, also info wont get logged unless embedded in the string

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 11]
```diff
@@ -1018,7 +1038,7 @@ export class FTSolanaWatcher extends SolanaWatcher {
     }
     try {
       const result = await this.pg('fast_transfer_auctions').where(id).update(update);
-      this.logger.info(`Updated auction ${id.fast_vaa_hash}. Result:`, result);
+      this.logger.info(`Updated auction ${id.fast_vaa_hash}. Result:`, update);
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:1041]: same comment as logging

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 12]
```diff
@@ -1079,10 +1109,22 @@ export class FTSolanaWatcher extends SolanaWatcher {
...
       const result = await this.pg('auction_logs').insert(auctionLogs);
-      this.logger.info(`Saved auction logs for ${auctionLogs.fast_vaa_hash}. Result:`, result);
+      this.logger.info(`Saved auction logs for ${auctionLogs.fast_vaa_hash}. Result:`, auctionLogs);
     } catch (error) {
       this.logger.error(`Error saving auction logs for ${auctionLogs.fast_vaa_hash}:`, error);
       throw error;
     }
   }
+
+  private handleMissingFastVaaHash(auctionPubkey: PublicKey, context: string): void {
```
Comment (bingyuyap) [inline watcher/src/watchers/FTSolanaWatcher.ts:1119]: add comment to explain

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 13]
```diff
@@ -688,7 +691,11 @@ export class FTSolanaWatcher extends SolanaWatcher {
     // if there is a auction to settle, the auction must exist. And the pubkey
     // will be in the db since we parse everything chronologically
     const fast_vaa_hash = await this.getFastVaaHashFromAuctionPubkey(auctionAccountPubkey.pubkey);
-
+    // We want to return the info even without the vaa hash if its a testing environment
```
Comment (panoel) [inline watcher/src/watchers/FTSolanaWatcher.ts:694]: Nit: its => it's

File: watcher/src/watchers/FTSolanaWatcher.ts [hunk 14]
```diff
@@ -1036,53 +1050,70 @@ export class FTSolanaWatcher extends SolanaWatcher {
...
+      this.logger.error(
+        `Error saving fast transfer info to table ${table}: ${error}, info: ${this.stringifyWithBigInt(
+          info
+        )}`
+      );
       throw error;
     }
   }
 
+  stringifyWithBigInt(obj: any) {
```
Comment (panoel) [inline watcher/src/watchers/FTSolanaWatcher.ts:1085]: Can you put this in a util file in common somewhere?  I've used something like this several times, too.

**Review by panoel — DISMISSED** [review]

**Review by panoel — APPROVED** [review]

**Pull request merged**